src/System.Management.Automation/engine/parser/Parser.cs

```diff
@@ -6609,7 +6725,11 @@ private ExpressionAst ArrayLiteralRule()
                 SkipToken();
                 SkipNewlines();
 
-                lastExpr = UnaryExpressionRule();
+                // We have seen a comma token and now expecting an expression as an array element.
+                // Allowing a generic token like '12?' or '12:' is not useful in the current situation,
```

Comment:
The comment is not exactly the same :) the first sentence is different for each of them.
It's not obvious to understand why `endNumberOnTernaryOpChars` is set to `true` for some of those method calls while `false` for others. I think having these comment here can help understand the code better.

Reply:
I agree about the comment aspect. Given that the first sentence changes but the rest doesn't, maybe worth considering the rest of the comment into the method (or its `remarks` field) itself?

Reply:
The rest of the comment can be made the comments for the parameter `endNumberOnTernaryOpChars`. I think that will be better, thought it will still be repeated 3 or 4 times.